modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBFilter.java

```diff
@@ -225,6 +225,25 @@ public MBFilter(JSONArray json, MBObjectParser parse, SemanticType semanticType)
         return Collections.emptySet();
     }
 
+    private Filter translateType(String jsonText) {
+        final FilterFactory2 ff = parse.getFilterFactory();
+        //TODO: How to wildcard geometry
+        Expression dimension = ff.function("dimension", ff.function("geometry"));
+
+        switch (jsonText) {
+            case "Point":
+                return ff.equals(dimension, ff.literal(0));
+            case "LineString":
+                return ff.equals(dimension, ff.literal(1));
+            case "Polygon":
```

Comment:
The spec does consistently refer to these with this casing, but I would vote for equalsIgnoreCase.